internal/webserver/wsauthconfig/config.go

```diff
@@ -0,0 +1,46 @@
+package wsauthconfig
+
+type OAuth2Conf struct {
+	ClientID     string   // OAuth client id (this app)
+	ClientSecret string   // OAuth2 secret (associated with ClientID, optional)
+	RedirectURL  string   // where should the OAuth2 provider return the user to
+	Scopes       []string // list of scopes to ask for from the OAuth2 provider
+}
+
+type OIDCConf struct {
+	IssuerURL string
+
+	// the ones below are only needed if the OIDC discovery mechanism doesn't work
+	AuthURL     string
+	TokenURL    string
+	UserInfoURL string
+	Algorithms  []string
+}
+
+// config for JWT token signature check
+type JWTConf struct {
+	UseJWKS bool
+	// used when UseJWKS is set to true
+	JwksURL              string
+	JwksSignatureMethods []string
+	// used when UseJWKS is set to false
+	Key           string // public key in case of asymmetric method, otherwise common secret (HMAC)
+	KeySignMethod string // can be "HS#", "RS#", "EC#", "EdDSA" (where # can be 256, 384, 512)
```

Comment:
There is also a "oneof" validator here that can be useful https://github.com/go-playground/validator

Reply:
see above